### ISSUE TITLE
About the accuracy of the baseline (SNGP)

### ISSUE THREAD

**Issue by johnny-richards** [y0ast/DUE#6]
Hi, I have successfully run this code and it performed well without "--sngp". However, when I added "sngp", the accuracy fell to 0.1. It might make sense when it was 0.7 or 0.8, but it seemed too low to be 0.1. 
Is it because I did something wrong, or the performance of GP layer is highly correlated with the feature extractor? Is such optimizer applicable for some bigger feature extractors, such as BERT for text classification for instant?
And by the way, is the performance significantly correlated with the optimizer? Is adam also okay?

**Comment by johnny-richards** [comment]
My bad, the sngp method does not incorporate dkl+gp

**Comment by johnny-richards** [comment]
Nevertheless I still encounter the problem that when I apply such model to a pretrained model like bert for classification, the loss always stucks at some high level and does not descent.

**Comment by y0ast** [comment]
Hey, there's a bunch of hyper-parameters you can try to change, but there's also an SNGP implementation specifically focused on text that seems to reproduce the paper!

https://github.com/kimjeyoung/SNGP-BERT-Pytorch

When running SNGP on CIFAR-10 I reproduce the paper using this repository.

**Comment by johnny-richards** [comment]
Cool, thank you. I'm going to try the SNGP-BERT repository.
